mdarea.js

```diff
@@ -257,7 +257,7 @@
     function handleInlineKey (elem, prefix, selection, postfix, key) {
         if (!selection && !(key in openingParens) && postfix.charAt(0) === key) {
             apply(elem, prefix + (reDoubledInline.test(key) ? key + key : '') + postfix, prefix.length + 1);
-        } else if (!selection && (key === "'" || key in closingParens)) {
+        } else if (!selection && (key === "'" && !/^[\s"'<>`\[\]\(\){}\+\*\^\$\\\.\|]$|^$/.test(prefix.slice(-1)) || key in closingParens)) {
```

Comment:
this looks good, only two suggestions here:
 - extract the regexp into a variable and put it at the start of the file with the rest of the regexps used throughout the code, call it something like `reSingleQuotePrefix` so that it's clear what its purpose is
 - add `=` and `-` to the regexp and remove some unnecessary backslashes
 - simplify the regexp and the expression it is used in to use the whole `prefix` - a single character regexp that is anchored at the end of the string is going to be fast and performant regardless of the input string lenght, there's no need to cut off the last character explicitly

I'd be happy to make these changes myself, I'm not a fan of rejecting PRs people put together in good faith or requesting that they remake them, but I'm not sure I can push commits into a PR that I don't own, do you have experience with this?

Reply:
Yep, if you clone my fork, check out the PRed branch, and push to it you should have access.